pkg/apiserver/handlers.go

```diff
@@ -66,6 +66,22 @@ func ReadOnly(handler http.Handler) http.Handler {
 	})
 }
 
+// MaxInFlight limits the number of in flight requests to the number of
+func MaxInFlightLimit(c chan bool, handler http.Handler) http.Handler {
+	if c == nil {
+		return handler
+	}
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		select {
+		case c <- true:
+			defer func() { <-c }()
+			handler.ServeHTTP(w, r)
+		default:
+			tooManyRequests(w)
+		}
```

Comment:
Is tooManyRequests logged as an error somewhere?  I assume so, but not sure where.  Just checking.